app/models/course/assessment/answer.rb

```diff
@@ -35,9 +35,13 @@ class Course::Assessment::Answer < ActiveRecord::Base
   def auto_grade!
     fail ArgumentError unless question.auto_gradable?
 
-    save!
-    create_auto_grading!
-    Course::Assessment::Answer::AutoGradingJob.new(auto_grading)
+    self.class.transaction do
+      save!
```

Comment:
http://apidock.com/rails/ActiveRecord/Validations/save%21

But save failure will give you a `RecordInvalid` exception, why we want this to be propagated ?